sql/core/src/test/java/test/org/apache/spark/sql/JavaBeanWithArraySuite.java

```diff
@@ -0,0 +1,222 @@
+                new StructField("endTime", DataTypes.LongType, true, Metadata.empty())
+        };
+        DataType intervalType = new StructType(intervalFields);
+
+        DataType intervalsType = new ArrayType(intervalType, true);
+
+        DataType valuesType = new ArrayType(DataTypes.IntegerType, true);
+
+        StructField[] fields = {
+                new StructField("id", DataTypes.IntegerType, true, Metadata.empty()),
+                new StructField("intervals", intervalsType, true, Metadata.empty()),
+                new StructField("values", valuesType, true, Metadata.empty())
+        };
+        return new StructType(fields);
+    }
+
+    public static class Record {
+
+        private int id;
+        private List<Interval> intervals;
+        private List<Integer> values;
+
+        public Record() { }
+
+        Record(int id, List<Interval> intervals, List<Integer> values) {
+            this.id = id;
+            this.intervals = intervals;
+            this.values = values;
+        }
+
+        public int getId() {
+            return id;
+        }
+
+        public void setId(int id) {
+            this.id = id;
+        }
+
+        public List<Interval> getIntervals() {
+            return intervals;
+        }
+
+        public void setIntervals(List<Interval> intervals) {
+            this.intervals = intervals;
+        }
+
+        public List<Integer> getValues() {
+            return values;
+        }
+
+        public void setValues(List<Integer> values) {
+            this.values = values;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (!(obj instanceof Record)) return false;
+            Record other = (Record) obj;
+            return
+                    (other.id == this.id) &&
+                    Util.equals(other.intervals, this.intervals) &&
+                    Util.equals(other.values, this.values);
+        }
+
+        @Override
+        public String toString() {
+            return String.format("{ id: %d, intervals: %s }", id, intervals );
+        }
+    }
+
+    public static class Interval {
+
+        private long startTime;
+        private long endTime;
+
+        public Interval() { }
+
+        Interval(long startTime, long endTime) {
+            this.startTime = startTime;
+            this.endTime = endTime;
+        }
+
+        public long getStartTime() {
+            return startTime;
+        }
+
+        public void setStartTime(long startTime) {
+            this.startTime = startTime;
+        }
+
+        public long getEndTime() {
+            return endTime;
+        }
+
+        public void setEndTime(long endTime) {
+            this.endTime = endTime;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (!(obj instanceof Interval)) return false;
+            Interval other = (Interval) obj;
+            return
```

Comment:
indentation?

Reply:
ah yes, in Spark we should use 2 space indention for java code as well